rpc/query_server.go

```diff
@@ -788,11 +789,9 @@ func (s *QueryServer) EthGetTransactionReceipt(hash eth.Data) (*eth.JsonTxReceip
 	if err != nil {
 		return nil, err
 	}
-	if int32(len(blockResult.Block.Data.Txs)) <= txReceipt.TransactionIndex {
-		return nil, errors.Errorf(
-			"Transaction index %v out of bounds for transactions in block %v",
-			txReceipt.TransactionIndex, len(blockResult.Block.Data.Txs),
-		)
+	txReceipt.TransactionIndex, err = s.getTransactionIndex(&txReceipt)
```

Comment:
What do you think about doing this only if `int32(len(blockResult.Block.Data.Txs)) <= txReceipt.TransactionIndex`?

Reply:
My worry is that the issue's due to the tx index being 1 based rather than zero-based. In which case, while you only get an error if the tx is the last in the block, your never going to return the right tx details.

Reply:
The tx index is 1-based?